### PR TITLE
Feanil/jenkins workers mongo update

### PR DESCRIPTION
**Pull request by feanil**
Configuration Pull Request
---

Make sure that the following steps are done before merging:

  - [ ] A DevOps team member has approved the PR.
  - [ ] Are you adding any new default values that need to be overridden when this change goes live? If so:
    - [ ] Update the appropriate internal repo (be sure to update for all our environments)
    - [ ] If you are updating a secure value rather than an internal one, file a DEVOPS ticket with details.
    - [ ] Add an entry to the CHANGELOG.
  - [ ] If you are making a complicated change, have you performed the proper testing specified on the [Ops Ansible Testing Checklist](https://openedx.atlassian.net/wiki/display/EdxOps/Ops+Ansible+Testing+Checklist)?  Adding a new variable does not require the full list (although testing on a sandbox is a great idea to ensure it links with your downstream code changes).

### CODE REVIEW

File: playbooks/roles/mongo_3_2/tasks/main.yml [hunk 1]
```diff
@@ -278,6 +278,7 @@
     rs_config: "{{ MONGO_RS_CONFIG }}"
   run_once: true
   register: replset_status
+  when: initialize_replica_set
```
Comment (jdmulloy) [inline playbooks/roles/mongo_3_2/tasks/main.yml]: These tasks are needed in situations that don't involve initializing a replica set, such as updating a replica set configuration or adding mongo users. Can we add a tag to the tasks that need to be skipped and use --skip-tags instead?

**Review by jdmulloy — CHANGES_REQUESTED** [review]

File: playbooks/roles/mongo_3_2/tasks/main.yml (hunk 1 above)
Comment (feanil) [reply playbooks/roles/mongo_3_2/tasks/main.yml]: @jdmulloy any suggestions on what to name that tag?

Comment (jdmulloy) [reply playbooks/roles/mongo_3_2/tasks/main.yml]: @feanil I don't have any great ideas. There is a wiki page of tags, but I haven't found it to be to helpful. https://openedx.atlassian.net/wiki/spaces/OpenOPS/pages/39584735/Role+Life-cycle+Tags

Maybe `skip_for_jenkins`, although I don't really like it.

Comment (feanil) [reply playbooks/roles/mongo_3_2/tasks/main.yml]: @jdmulloy looking more closely at the tasks that I am adding the when condition to.  All of them expect the replica set to be created or don't work.  Thinking back, that is why I had to add the when condition to these tasks in the first place.

File: util/packer/jenkins_worker.json [hunk 2]
```diff
@@ -25,7 +25,13 @@
     "security_group_id": "{{user `security_group`}}",
     "tags": {
       "delete_or_keep": "{{user `delete_or_keep`}}"
```
Comment (estute) [inline util/packer/jenkins_worker.json:28]: @feanil I think you will need to expose JENKINS_WORKER_KEY_URL in the packer script as an environment variable

**Review by estute — CHANGES_REQUESTED** [review]
See comment

File: util/packer/jenkins_worker.json (hunk 2 above)
Comment (feanil) [reply util/packer/jenkins_worker.json:28]: I had forgotten to push up that commit.  @estute ready for re-review.

**Review by estute — APPROVED** [review]

**Comment by feanil** [comment]
@jdmulloy I think this is ready for a final review.

**Review by jdmulloy — APPROVED** [review]

**Pull request merged**